server/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -199,34 +197,29 @@ public void indexBlockedRaiseException(ClusterBlockLevel level, String index) th
     }
 
     public ClusterBlockException indexBlockedException(ClusterBlockLevel level, String index) {
-        if (!indexBlocked(level, index)) {
-            return null;
-        }
-        Stream<ClusterBlock> blocks = concat(
-                global(level).stream(),
-                blocksForIndex(level, index).stream());
-        return new ClusterBlockException(unmodifiableSet(blocks.collect(toSet())));
+        return indicesBlockedException(level, new String[]{index});
     }
 
     public boolean indexBlocked(ClusterBlockLevel level, String index) {
         return globalBlocked(level) || blocksForIndex(level, index).isEmpty() == false;
     }
 
     public ClusterBlockException indicesBlockedException(ClusterBlockLevel level, String[] indices) {
-        boolean indexIsBlocked = false;
+        Set<ClusterBlock> globalLevelBlocks = global(level);
+        Map<String, Set<ClusterBlock>> indexLevelBlocks = new HashMap<>();
         for (String index : indices) {
-            if (indexBlocked(level, index)) {
-                indexIsBlocked = true;
+            Set<ClusterBlock> indexBlocks = blocksForIndex(level, index);
+            if (indexBlocks.isEmpty() == false || globalLevelBlocks.isEmpty() == false) {
+                indexLevelBlocks.put(index, Sets.union(indexBlocks, globalLevelBlocks));
             }
         }
-        if (globalBlocked(level) == false && indexIsBlocked == false) {
+        if (indexLevelBlocks.isEmpty()) {
+            if(!globalLevelBlocks.isEmpty()){
```

Comment:
Can we please change this to `globalLevelBlocks.isEmpty() == false` as we prefer "pred == false` which is easier to read than `!pred`.